### PR TITLE
Change ip vrf for router OS v7

### PR DESCRIPTION
**Pull request by liquorice-head**
##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below
In router OS 7, the path to vrf and the fields used have changed
tested on v7.13.3
```

### CODE REVIEW

**Review by derdeagle — CHANGES_REQUESTED** [review]

File: plugins/module_utils/_api_data.py [hunk 1]
```diff
@@ -727,15 +727,16 @@ def join_path(path):
...
             fully_understood=True,
-            primary_keys=('routing-mark', ),
+            primary_keys=('name', ),
             fields={
                 'comment': KeyInfo(can_disable=True, remove_value=''),
                 'disabled': KeyInfo(default=False),
                 'interfaces': KeyInfo(),
-                'routing-mark': KeyInfo(),
+                'name': KeyInfo(),
+                'numbers': KeyInfo(),
```
[the remark this answers is not in the document]
Comment (liquorice-head) [reply plugins/module_utils/_api_data.py:739]: Screenshot 2024-02-06 at 15 00 26
in ip vrf set there is such an entry

File: plugins/module_utils/_api_data.py [hunk 2]
```diff
@@ -727,15 +727,16 @@ def join_path(path):
             },
         ),
     ),
-    ('ip', 'route', 'vrf'): APIData(
+    ('ip', 'vrf'): APIData(
```
Comment (felixfontein) [inline plugins/module_utils/_api_data.py:730]: This breaks all users of older RouterOS versions. Better add a copy of this path for RouterOS 7 with the new path, and keep the old one for previous versions.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Below a first comment from my side. Please note that you need to add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment), but it might be easier to wait a bit until it's clear how this PR will look before adding it :)

File: plugins/module_utils/_api_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liquorice-head) [reply plugins/module_utils/_api_data.py]: Thanks for the explanation. I've made the change.

**Comment by ialshaev** [comment]
@felixfontein is it possible to have both versions for vrf path? So we can decide what path to chose depending on RouterOS version.
Thank you in advance!

**Comment by liquorice-head** [comment]
Added versioning. Pls review

File: plugins/modules/api_info.py [hunk 3]
```diff
@@ -136,6 +136,7 @@
         - ip proxy
         - ip route
         - ip route vrf
+        - ip vrf
```
Comment (felixfontein) [inline plugins/modules/api_info.py:139]: This needs to be moved to the right position (lexicographically sorted), i.e. after `ip upnp interfaces`. Same in the other file.

**Review by felixfontein — COMMENTED** [review]
Looks almost good now, see my comment below! Please don't forget the [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment) :)

**Comment by liquorice-head** [comment]
Done. Please review

File: changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - api_data - add support for the ``ip vrf`` path in RouterOS 7 and add support new primary_keys ``routing-mark`` for RouterOS 7 (https://github.com/ansible-collections/community.routeros/pull/259)
+  - api_modify, api_info - add support for the ``ip vrf`` path in RouterOS 7  (https://github.com/ansible-collections/community.routeros/pull/259)
```
Comment (felixfontein) [inline changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml:3]:
```suggestion
  - api_modify, api_info - add support for the ``ip vrf`` path in RouterOS 7 and support new primary keys ``routing-mark`` for RouterOS 7 (https://github.com/ansible-collections/community.routeros/pull/259)
```

File: changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - api_data - add support for the ``ip vrf`` path in RouterOS 7 and add support new primary_keys ``routing-mark`` for RouterOS 7 (https://github.com/ansible-collections/community.routeros/pull/259)
```
Comment (felixfontein) [inline changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml:2]:
```suggestion
```

**Review by derdeagle — APPROVED** [review]
Looks good to me.

File: changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml (hunk 4 above)
Comment (felixfontein) [reply changelogs/fragments/259-add-routeros7-support-for-ip-vrf.yml:2]: I copied that part over from the other part that was removed. If it's wrong, I don't mind not having this here :)

**Comment by felixfontein** [comment]
@liquorice-head thanks for implementing this!
@derdeagle thanks for reviewing this!

**Comment by derdeagle** [comment]
Always a pleasure ;)

**Pull request merged**